File: macos/QMK Toolbox/USB/Bootloader/STM32DuinoDevice.m
#import "STM32DuinoDevice.h"

@implementation STM32DuinoDevice

- (id)initWithUSBDevice:(USBDevice *)usbDevice {
    if (self = [super initWithUSBDevice:usbDevice]) {
        self.name = @"STM32Duino";
        self.type = BootloaderTypeSTM32Duino;
    }
    return self;
}

-(void)flashWithMCU:(NSString *)mcu file:(NSString *)file {
    if([[[file pathExtension] lowercaseString] isEqualToString:@"bin"]) {
        [self runProcess:@"dfu-util" withArgs:@[@"-a", @"2", @"-d", @"1EAF:0003", @"-R", @"-D", file]];
    } else {
        [self printMessage:@"Only firmware files in .bin format can be flashed with dfu-util!" withType:MessageType_Error];
    }
}

@end

File: macos/QMK Toolbox/USB/Bootloader/STM32DuinoDevice.swift
import Foundation

class STM32DuinoDevice: BootloaderDevice {
    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "STM32Duino"
        type = .stm32duino
    }

    override func flash(_ mcu: String, file: String) {
        guard file.lowercased().hasSuffix(".bin") else {
            print(message: "Only firmware files in .bin format can be flashed with dfu-util!", type: .error)
            return
        }

        runProcess("dfu-util", args: ["-a", "2", "-d", "1EAF:0003", "-R", "-D", file])
    }
}
